components/brave_wallet/browser/solana_tx_manager.cc

```diff
@@ -361,6 +366,39 @@ void SolanaTxManager::MakeTokenProgramTransferTxData(
           std::move(callback)));
 }
 
+void SolanaTxManager::MakeTokenProgramTxDataFromMessage(
+    const std::string& message,
+    const mojom::TransactionType tx_type,
+    mojom::SolanaSendTransactionOptionsPtr send_options,
+    MakeTokenProgramTxDataFromMessageCallback callback) {
+  absl::optional<std::vector<std::uint8_t>> message_bytes =
+      base::Base64Decode(message);
+  if (!message_bytes || message_bytes->empty() ||
+      message_bytes->size() > kSolanaMaxTxSize) {
+    std::move(callback).Run(
+        nullptr, mojom::SolanaProviderError::kInternalError,
+        l10n_util::GetStringUTF8(IDS_WALLET_INTERNAL_ERROR));
+    return;
+  }
+
+  auto transaction =
+      SolanaTransaction::FromSignedTransactionBytes(*message_bytes);
```

Comment:
This shouldn't be used if what you have is just message bytes. This one is for signed transactions. You should use SolanaMessage::Deserialize and create your transaction from it.

Reply:
The base64-encoded string provided by Jupiter is actually signed transaction bytes (compact-array of signatures followed by message). In this particular case, the signature is empty (all zeros). I think the reason is that Jupiter expects clients to do the following:

```js
import { Transaction } from '@solana/web3.js'
Transaction.from(Buffer.from(serializedTransaction, 'base64'))
```

With that in mind, maybe we should rename our function to `MakeTxDataFromBase64EncodedTranasction`.

Reply:
Added here: de66ca07d77ca57ac8f2862a1801fdfcb7e583ee